mac80211: ath11k: fix rssi for ipq5018 and qcn6122

TheRootED24 opened the pull request:
references:
>upstream commit: [031ffa6c2cd305a57ccc6d610f2decd956b2e7f6] 
Subject: [PATCH] wifi: ath11k: fix rssi station dump not updated in QCN9074

symptoms: iw station dump displays default value for avg signal (-95dbm) and signal (0dbm)

As described in the refrenced upstream commit, there was a header change
for HAL_RX_MPDU_START between ipq8074 and qcn9074. This caused wrong peer_id fetch
from msdu in the 9074 and the ipq5018 and subsequently the qcn6122 which uses the
ipq5018 hw_opts.

solution: use the updated qcn9074 mpdu_info_get_peerid implimentation with the new tlv
format, from the referenced upstream commit.

TheRootED24 commented:
here is what iw station dump returned before applying patch

```
 root@OpenWrt:~# iw phy1-ap0 station dump
Station xx:xx:xx:xx:xx:ba (on phy1-ap0)
        inactive time:  71940 ms
        rx bytes:       18542777
        rx packets:     47145
        tx bytes:       340746485
        tx packets:     282365
        tx retries:     5139
        tx failed:      1275
        rx drop misc:   0
        signal:         0 dBm
        signal avg:     -95 dBm
        tx bitrate:     6.0 MBit/s
        tx duration:    10892904 us
        rx bitrate:     48.0 MBit/s
        rx duration:    2389771 us
        last ack signal:-67 dBm
        avg ack signal: -64 dBm
        authorized:     yes
        authenticated:  yes
        associated:     yes
        preamble:       short
        WMM/WME:        yes
        MFP:            no
        TDLS peer:      no
        DTIM period:    2
        beacon interval:100
        short preamble: yes
        short slot time:yes
        connected time: 7486 seconds
        associated at [boottime]:       224.166s
        associated at:  1763589325064 ms
        current time:   1763596811543 ms

```

after patch applied 

```
Station xx:xx:xx:xx:xx:ba (on phy1-ap0)
        inactive time:  145490 ms
        rx bytes:       18542777
        rx packets:     47145
        tx bytes:       340746485
        tx packets:     282365
        tx retries:     5139
        tx failed:      1275
        rx drop misc:   0
        signal:         -59 dBm
        signal avg:     -61 dBm
        tx bitrate:     6.0 MBit/s
        tx duration:    10893784 us
        rx bitrate:     48.0 MBit/s
        rx duration:    2390194 us
        last ack signal:-71 dBm
        avg ack signal: -68 dBm
        authorized:     yes
        authenticated:  yes
        associated:     yes
        preamble:       short
        WMM/WME:        yes
        MFP:            no
        TDLS peer:      no
        DTIM period:    2
        beacon interval:100
        short preamble: yes
        short slot time:yes
        connected time: 7560 seconds
        associated at [boottime]:       224.166s
        associated at:  1763589325022 ms
        current time:   1763596885095 ms

```

Luci SNR values we not populating either. There is an open ticket for it
https://github.com/openwrt/openwrt/issues/18563#issuecomment-3544020154

the patch fixes that issue as well, luci now fully populates all fields correctly after applying patch

castiel652 commented:
Can you send this to upstream

castiel652 commented:
https://wireless.docs.kernel.org/en/latest/en/users/drivers/ath11k/submittingpatches.html
Read this

If you have more questions feel free to ask. :)

castiel652 commented on package/kernel/mac80211/patches/ath11k/947-ath11k-ipq50xx-fix-rssi.patch:
yes normal.
I assume you didn't make the patch with `make package/kernel/mac80211/{clean,prepare} V=s QUILT=1`

castiel652 commented:
commit title prefix should be `mac80211: ath11k:`

psyke83 commented:
Confirming that local build with latest push (ab94f612e859f09f75c5cfa11f664284696ab680) populates RSSI correctly for both radios of GL-B3000.

TheRootED24 committed:
mac80211: ath11k: fix rssi for IPQ5018 and QCN6122

Add pending patch to fix RSSI station data on IPQ5018 and QCN6122.

Signed-off-by: Scott Mercer <TheRootEd24@gmail.com>
[ improve commit description, replace patch ]
Link: https://github.com/openwrt/openwrt/pull/20834
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

georgemoussalem commented on package/kernel/mac80211/patches/ath11k/947-ath11k-ipq50xx-fix-rssi.patch:
```diff
@@ -0,0 +1,25 @@
+
+solution: use the updated qcn9074 mpdu_info_get_peerid implimentation with the new tlv
+format, from the referenced upstream commit.
+
```
the patch must contain your signed-off-by tag:

`Signed-off-by: Scott Mercer <TheRootEd24@gmail.com>`

castiel652 commented:
https://lore.kernel.org/linux-wireless/20251125142821.18152-1-ansuelsmth@gmail.com/

Ansuel commented:
It Will eventually and trust me QCOM is aware of the similarities

Il Gio 27 Nov 2025, 03:10 Zhi-Jun You ***@***.***> ha scritto:

> *castiel652* left a comment (openwrt/openwrt#20834)
> <https://github.com/openwrt/openwrt/pull/20834#issuecomment-3583937691>
>
> also just realised something.
> @Ansuel <https://github.com/Ansuel> QCN6122 is not supported in upstream
> so you might need to omit it from the patch.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/20834#issuecomment-3583937691>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQT7HOE7H4MS6IVWXLT36ZMRHAVCNFSM6AAAAACMUHICHCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKOBTHEZTONRZGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
